ui/src/app/applications/components/revision-form-field/revision-form-field.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {FormApi} from 'react-form';
+
+import {AutocompleteField, DataLoader, FormField} from 'argo-ui';
+import {RevisionHelpIcon} from '../../../shared/components';
+import {services} from '../../../shared/services';
+
+interface RevisionFormFieldProps {
+    formApi: FormApi;
+    helpIconTop?: string;
+    hideLabel?: boolean;
+    repoURL: string;
+}
+
+export class RevisionFormField extends React.PureComponent<RevisionFormFieldProps> {
+    public render() {
+        return (
+            <React.Fragment>
+                <DataLoader
+                    input={{repoURL: this.props.repoURL}}
+                    load={async (src: any): Promise<string[]> => {
+                        if (src.repoURL) {
+                            try {
+                                const revisionsRes = await services.repos.revisions(src.repoURL);
```

Comment:
I think it would be easy to read if we don't use async. E.g.

```typescript
return services.repos.revisions(src.repoURL).
  then(revisionsRes =>[ 'HEAD' ].concat(revisionsRes.branches).concat(revisionsRes.tags) ).
  catch(() => []);
```

What do you think?

Reply:
sure, doesn't matter